sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -562,13 +552,17 @@ private boolean checkDoesNotExistStatusCode(Throwable t) {
 
         if (filePermissionCopyMode == null || filePermissionCopyMode == PermissionCopyModeType.SOURCE) {
             if (filePermission != null || filePermissionKey != null) {
-                throw logger.logExceptionAsError(new IllegalArgumentException(
+                return PollerFlux.error(LOGGER.logExceptionAsError(new IllegalArgumentException(
                     "File permission and file permission key can not be set when PermissionCopyModeType is source or "
-                        + "null"));
+                        + "null")));
             }
         } else if (filePermissionCopyMode == PermissionCopyModeType.OVERRIDE) {
             // Checks that file permission and file permission key are valid
-            validateFilePermissionAndKey(filePermission, tempSmbProperties.getFilePermissionKey());
+            try {
+                validateFilePermissionAndKey(filePermission, tempSmbProperties.getFilePermissionKey());
+            } catch (RuntimeException ex) {
+                return PollerFlux.error(LOGGER.logExceptionAsError(ex));
+            }
```

Comment:
would it be better to make validateFilePermissionAndKey return Mono instead of throwing ? Which is more expensive?

Reply:
I think it would but the method is called from APIs that return Mono and PagedFlux